### DISCUSSION TITLE
What will happen if i change useragent

### DISCUSSION THREAD

**Discussion by ysfdumlu** [streamlink/streamlink#5290] [Q&A] [answered]
### Description

if i change `useragent.IPHONE_6` to IPHONE_14 for example what will happen ?

I think maybe it is possible that the stream server could send a higher quality stream to an iPhone 14 compared to an iPhone 6. Because the iPhone 14 has a higher resolution screen compared to the iPhone 6.
Of course i change it and tried  already :) But it is not that easy as i expected. it gives multiple errors such as:
```
File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Program Files\Streamlink\bin\streamlink.exe\__main__.py", line 18, in <module>
  File "C:\Program Files\Streamlink\pkgs\streamlink_cli\main.py", line 935, in main
    handle_url()
  File "C:\Program Files\Streamlink\pkgs\streamlink_cli\main.py", line 552, in handle_url
    streams = fetch_streams_with_retry(plugin, retry_streams, retry_max)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Streamlink\pkgs\streamlink_cli\main.py", line 461, in fetch_streams_with_retry
    streams = fetch_streams(plugin)
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Streamlink\pkgs\streamlink_cli\main.py", line 452, in fetch_streams
    return plugin.streams(stream_types=args.stream_types,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Streamlink\pkgs\streamlink\plugin\plugin.py", line 428, in streams
    ostreams = list(ostreams)
               ^^^^^^^^^^^^^^
  File "C:\Program Files\Streamlink\pkgs\streamlink\plugins\bigo.py", line 33, in _get_streams
    headers={"User-Agent": useragents.IPHONE_14},
                           ^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'streamlink.plugin.api.useragents' has no attribute 'IPHONE_14'. Did you mean: 'IPHONE_6'?
```

If it had given the errors one by one, I would have tried to correct them one by one. but when it gave errors in many places, I just wanted to ask if it would be the result I expected or is it irrelevant to change the useragent.
![usera](https://user-images.githubusercontent.com/106635100/231907249-17fdfe1d-8a6a-4972-85c7-faa702d715e3.jpg)

**Comment by bastimeyer** [comment, marked as answer]
> `useragent.IPHONE_6`

This is a constant from the `useragent` module. You can't just set any arbitrary constants that don't exist. This is not how this works.
https://github.com/streamlink/streamlink/blob/master/src/streamlink/plugin/api/useragents.py

And btw, the bigo plugin sets the iphone useragent, because years ago, the site served flash streams, and setting the user agent to iphone allowed accessing HLS streams, as you can see in the following commit derived from the git-blame of the plugin file:
5303e40bed009ffd3ca6ef27697259197e12fe72

**Reply by ysfdumlu** [reply]
With chatgpt, you can go up to a point. :) Thank you for your quick answer.

Regardless of this topic, I tried something interesting to me. While I was wondering why streamlink doesn't have a plugin for tiktok livestream, I tried to print the plugin on 2 different artificial intelligence models. One of them is of course chatgpt and the other is "zekai.co" (actually the same model, but gives different answers.) First, take the printout of one of them and say to the other "analyze this code and give your suggestions", then "rewrite it taking your suggestions into account" repeatedly. Since I do not have a very serious level of knowledge, I proceed by trial and error. However, when I review the already existing written plugins on github, it gives very similar things. Of course, I know that even a letter or punctuation mark will cause it to not work, but I am a little interested in this work as an experimental curiosity.
